src/browser/cdp/selectivity/json-utils.ts

```diff
@@ -0,0 +1,158 @@
+import fs from "node:fs";
+import zlib from "node:zlib";
+import type { SelectivityCompressionType } from "./types";
+import type { Readable, Transform, Writable } from "node:stream";
+
+const getCompressionPrefix = (type: SelectivityCompressionType): string => (type === "none" ? "" : "." + type);
+
+const compressionPriorities: SelectivityCompressionType[] = process.versions.zstd
+    ? ["zstd", "br", "gz", "none"]
+    : ["br", "gz", "none"];
+
+const getExistingJsonPathWithCompression = (
+    jsonBasePath: string,
+    preferredCompressionType: SelectivityCompressionType,
+): { jsonPath: string | null; compressionType: SelectivityCompressionType } => {
+    const jsonPathWithPrefix = jsonBasePath + getCompressionPrefix(preferredCompressionType);
+
+    if (fs.existsSync(jsonPathWithPrefix)) {
+        return { jsonPath: jsonPathWithPrefix, compressionType: preferredCompressionType };
+    }
+
+    for (const compressionType of compressionPriorities) {
+        if (compressionType === preferredCompressionType) {
+            continue;
+        }
+
+        const jsonPathWithCurrentPrefix = jsonBasePath + getCompressionPrefix(compressionType);
+        if (fs.existsSync(jsonPathWithCurrentPrefix)) {
+            return { jsonPath: jsonPathWithCurrentPrefix, compressionType: compressionType };
+        }
+    }
+
+    return { jsonPath: null, compressionType: preferredCompressionType };
+};
+
+const readCompressedTextFile = (filePath: string, compression: SelectivityCompressionType): Promise<string> => {
+    return new Promise<string>((resolve, reject) => {
+        let fileData = "";
+        const readStream = fs.createReadStream(filePath);
+        let stream: Readable = readStream;
+
+        if (compression === "zstd" && !("createZstdDecompress" in zlib)) {
+            reject(new Error("zstd decompression is not supported in your version of node.js. Please upgrade to 22"));
+        }
+
+        switch (compression) {
+            case "zstd":
```

Comment:
Can we create an enum for these values? I see a lot of places where we use these strings.